packages/form-fields/text/index.tsx

```diff
@@ -0,0 +1,63 @@
+import classNames from 'classnames'
+import { useId } from '@reach/auto-id'
+import s from './style.module.css'
+import type { HTMLProps } from 'react'
+import Label from '../partials/label'
+
+interface TextInputProps {
+  className?: string
+  field: HTMLProps<HTMLInputElement>
+  error?: string
+  type: HTMLProps<HTMLInputElement>['type']
```

Comment:
@dstaley can we make `type` optional since its [default is defined](https://github.com/hashicorp/react-components/pull/801/files#diff-675d87d14ef79b60061d78f4ece576cd345e68fbeae562cbceb71c5a9beaa068R23)?